### PR TITLE
docs: setup unified GTM

### PR DESCRIPTION
**Pull request by saihaj**
## Why are these changes needed?

A unified setup with Google Tag manager to manage scripts that are loaded and also respecting user cookie configuration.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/conf.py [hunk 1]
```diff
@@ -288,7 +288,6 @@ def render_svg_logo(path):
         "csat",
     ],
     "navigation_depth": 4,
-    "analytics": {"google_analytics_id": "UA-110413294-1"},
```
Comment (can-anyscale) [inline doc/source/conf.py:291]: n00b question, why we are removing google analytics here?

Comment (saihaj) [reply doc/source/conf.py:291]: that will be injected by [Google Tag manager](https://developers.google.com/tag-platform/tag-manager). It is to manage all tags in one place for all websites we manage. With Google tag manager we will also inject cookiebot which will make sure Google analytics only runs when user gives permissions (where needed for compliance reasons)

**Review by cristianjd — APPROVED** [review]

**Review by can-anyscale — CHANGES_REQUESTED** [review]
thanks for the PR; it conflicts with the master branch now; if you can help resolve i'll merge afterwards, thankks

**Review by can-anyscale — APPROVED** [review]

**Pull request merged**